dispatch/apps/api/views.py

```diff
@@ -407,6 +407,42 @@ def articles(self, request, pk=None):
         view = ArticleViewSet.as_view({'get': 'topic'})
         return view(request, pk=pk)
 
+class FileViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for File model views.
+    """
+    model = File
+    serializer_class = FileSerializer
+
+    def get_queryset(self):
+        queryset = File.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            # If a search term (q) is present, filter queryset by term against `name`
+            queryset = queryset.filter(name__icontains=q)
+        return queryset
+
+    def bulk_delete(self, request):
+        deleted = []
+        ids = self.request.data.get('ids', None)
```

Comment:
Most meaningful would probably be to remove the bulk endpoint, unless it's a real pain point for the frontend.

As it is currently, you're building a queryset (filter(IDs in `ids`)) and then calling delete on it -- that should never fail. If the frontend passes bad IDs, you'll just get an empty qs, which it's still valid to call delete on.

Apparently since django 1.3 deleting a model that has a FileField doesn't actually delete the file itself anyway, due to concerns about losing data in transaction rollbacks. ([django docs](https://docs.djangoproject.com/en/1.9/releases/1.3/#deleting-a-model-doesn-t-delete-associated-files))

There's some answers on [SO](http://stackoverflow.com/questions/16041232/django-delete-filefield) and a link to this apparently unmaintained [library](https://github.com/un1t/django-cleanup).